### ISSUE TITLE
Task `1cfdca43-873a-4b36-ab42-7baf655d72c9` is corrupted

### ISSUE THREAD

**Issue by benoit74** [openzim/zimfarm#824]
Location: API

On 22. sept 2023, we have 14 "PATCH 499" errors, all linked to the same task [1cfdca43-873a-4b36-ab42-7baf655d72c9](https://farm.openzim.org/pipeline/1cfdca43-873a-4b36-ab42-7baf655d72c9):

Let's trace the history (all times are UTC, container pings are omitted for brevity):
- At `2023-09-22T05:00:02`, task is requested by periodic-scheduler
- At `2023-09-22T05:05:36`, chance worker is requesting a new task but scheduler decides that there is insufficient memory for the first task (not our task) on the pile 
- At `2023-09-22T05:08:39` chance is assigned a task, probably our "culprit" since it is reserved just after ; other tasks have been assigned to other workers in the mean time
- At `2023-09-22T05:08:39`, task is reserved by the worker
- At `2023-09-22T05:08:46`, task is started
- At `2023-09-22T05:31:05`, task has created file: wikisource_id_all_nopic_2023-09.zim, 111203786
- At `2023-09-22T05:32:19.234`, task has checked file: wikisource_id_all_nopic_2023-09.zim
  - At `2023-09-22 07:32:19.734`, we try to contact the CMS (not sure because we have concurrent task updates, but it seems logical to be linked to our task)
  - At `2023-09-22 07:32:19.985` we have an exception because we fail to parse the CMS response, and we have another exception during exception handling' which causes a fatal issue
  - At `2023-09-22 07:33:19.235`, a 499 error is replied
    - **why not before?** due to the exception during exception handling?
  - From this point, all subsequent PATCH requests on this task are failing either with 499 or 504 errors ; GET requests on this task are responding properly

Below the list of HTTP events:
```
2023-09-22T05:08:39.702244841Z stdout F 100.64.6.28 - - [22/Sep/2023:05:08:39 +0000] "POST /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9?worker_name=chance HTTP/1.1" 201 2180 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:08:40.19233546Z stdout F 100.64.6.28 - - [22/Sep/2023:05:08:40 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 934 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:08:43.782857483Z stdout F 100.64.6.28 - - [22/Sep/2023:05:08:43 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 934 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:08:44.302132847Z stdout F 100.64.6.28 - - [22/Sep/2023:05:08:44 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:08:46.45898377Z stdout F 100.64.6.28 - - [22/Sep/2023:05:08:46 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:08:47.989981304Z stdout F 100.64.6.28 - - [22/Sep/2023:05:08:47 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:09:49.686436428Z stdout F 100.64.6.28 - - [22/Sep/2023:05:09:49 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:10:50.404446343Z stdout F 100.64.6.28 - - [22/Sep/2023:05:10:50 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:11:51.076918656Z stdout F 100.64.6.28 - - [22/Sep/2023:05:11:51 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:12:51.746592566Z stdout F 100.64.6.28 - - [22/Sep/2023:05:12:51 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:13:13.671548923Z stdout F 100.64.6.28 - - [22/Sep/2023:05:13:13 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 1473 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:13:52.469266693Z stdout F 100.64.6.28 - - [22/Sep/2023:05:13:52 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:14:53.18130414Z stdout F 100.64.6.28 - - [22/Sep/2023:05:14:53 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:15:53.886419091Z stdout F 100.64.6.28 - - [22/Sep/2023:05:15:53 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:16:16.495182795Z stdout F 100.64.6.28 - - [22/Sep/2023:05:16:16 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 1587 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:16:54.581551729Z stdout F 100.64.6.28 - - [22/Sep/2023:05:16:54 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:17:55.378139126Z stdout F 100.64.6.28 - - [22/Sep/2023:05:17:55 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:18:56.18893338Z stdout F 100.64.6.28 - - [22/Sep/2023:05:18:56 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:19:19.413775581Z stdout F 100.64.6.28 - - [22/Sep/2023:05:19:19 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 1705 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:19:56.876887122Z stdout F 100.64.6.28 - - [22/Sep/2023:05:19:56 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:20:57.569496626Z stdout F 100.64.6.28 - - [22/Sep/2023:05:20:57 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:21:58.450619885Z stdout F 100.64.6.28 - - [22/Sep/2023:05:21:58 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:22:22.298520844Z stdout F 100.64.6.28 - - [22/Sep/2023:05:22:22 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 1828 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:22:59.162830779Z stdout F 100.64.6.28 - - [22/Sep/2023:05:22:59 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:23:59.897875767Z stdout F 100.64.6.28 - - [22/Sep/2023:05:23:59 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:25:00.65418593Z stdout F 100.64.6.28 - - [22/Sep/2023:05:25:00 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:25:25.302294674Z stdout F 100.64.6.28 - - [22/Sep/2023:05:25:25 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 1897 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:26:01.391560371Z stdout F 100.64.6.28 - - [22/Sep/2023:05:26:01 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:27:02.061903604Z stdout F 100.64.6.28 - - [22/Sep/2023:05:27:02 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:28:02.736365771Z stdout F 100.64.6.28 - - [22/Sep/2023:05:28:02 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:28:28.253624815Z stdout F 100.64.6.28 - - [22/Sep/2023:05:28:28 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 1983 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:29:03.435508451Z stdout F 100.64.6.28 - - [22/Sep/2023:05:29:03 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:30:04.11343264Z stdout F 100.64.6.28 - - [22/Sep/2023:05:30:04 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:31:04.807462698Z stdout F 100.64.6.28 - - [22/Sep/2023:05:31:04 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:31:05.434508322Z stdout F 100.64.6.28 - - [22/Sep/2023:05:31:05 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:31:31.151350882Z stdout F 100.64.6.28 - - [22/Sep/2023:05:31:31 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 2350 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:32:05.232046152Z stdout F 100.64.6.28 - - [22/Sep/2023:05:32:05 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 204 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:32:19.019455001Z stderr F 2023/09/22 05:32:19 [warn] 15#15: *33528 a client request body is buffered to a temporary file /var/cache/nginx/client_temp/0000002389, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", host: "api.farm.openzim.org"
2023-09-22T05:33:19.016735396Z stdout F 100.64.6.28 - - [22/Sep/2023:05:33:19 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:34:20.539623728Z stdout F 100.64.6.28 - - [22/Sep/2023:05:34:20 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:34:34.029231279Z stdout F 100.64.6.28 - - [22/Sep/2023:05:34:34 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 2386 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:35:21.019316513Z stderr F 2023/09/22 05:35:21 [error] 15#15: *33847 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "api.farm.openzim.org"
2023-09-22T05:35:21.019429535Z stdout F 100.64.6.28 - - [22/Sep/2023:05:35:21 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 504 167 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:36:22.549778579Z stdout F 100.64.6.28 - - [22/Sep/2023:05:36:22 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:37:24.051247571Z stderr F 2023/09/22 05:37:24 [error] 15#15: *33883 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "api.farm.openzim.org"
2023-09-22T05:37:24.051325069Z stdout F 100.64.6.28 - - [22/Sep/2023:05:37:24 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 504 167 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:37:36.908890722Z stdout F 100.64.6.28 - - [22/Sep/2023:05:37:36 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 2386 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:38:25.585817509Z stdout F 100.64.6.28 - - [22/Sep/2023:05:38:25 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:39:27.08973455Z stdout F 100.64.6.28 - - [22/Sep/2023:05:39:27 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:40:28.607291079Z stderr F 2023/09/22 05:40:28 [error] 15#15: *33908 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "api.farm.openzim.org"
2023-09-22T05:40:28.607393687Z stdout F 100.64.6.28 - - [22/Sep/2023:05:40:28 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 504 167 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:40:39.783012381Z stdout F 100.64.6.28 - - [22/Sep/2023:05:40:39 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 2386 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:41:30.241969402Z stdout F 100.64.6.28 - - [22/Sep/2023:05:41:30 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:42:31.761705371Z stdout F 100.64.6.28 - - [22/Sep/2023:05:42:31 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:43:33.26383589Z stderr F 2023/09/22 05:43:33 [error] 15#15: *34204 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "api.farm.openzim.org"
2023-09-22T05:43:33.263933182Z stdout F 100.64.6.28 - - [22/Sep/2023:05:43:33 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 504 167 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:43:42.674115273Z stdout F 100.64.6.28 - - [22/Sep/2023:05:43:42 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 2386 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:44:34.777874688Z stderr F 2023/09/22 05:44:34 [error] 15#15: *34120 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "api.farm.openzim.org"
2023-09-22T05:44:34.778034091Z stdout F 100.64.6.28 - - [22/Sep/2023:05:44:34 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 504 167 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:45:36.339241937Z stderr F 2023/09/22 05:45:36 [error] 15#15: *34192 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "api.farm.openzim.org"
2023-09-22T05:45:36.339431464Z stdout F 100.64.6.28 - - [22/Sep/2023:05:45:36 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 504 167 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:46:37.847588675Z stdout F 100.64.6.28 - - [22/Sep/2023:05:46:37 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:46:45.544227561Z stdout F 100.64.6.28 - - [22/Sep/2023:05:46:45 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 2386 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:47:39.34960544Z stdout F 100.64.6.28 - - [22/Sep/2023:05:47:39 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:48:40.856400795Z stdout F 100.64.6.28 - - [22/Sep/2023:05:48:40 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:49:42.37382988Z stdout F 100.64.6.28 - - [22/Sep/2023:05:49:42 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:49:48.700180132Z stdout F 100.64.6.28 - - [22/Sep/2023:05:49:48 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 2386 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:50:43.87377524Z stdout F 100.64.6.28 - - [22/Sep/2023:05:50:43 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:51:45.885256534Z stderr F 2023/09/22 05:51:45 [error] 15#15: *34459 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "api.farm.openzim.org"
2023-09-22T05:51:45.885376985Z stdout F 100.64.6.28 - - [22/Sep/2023:05:51:45 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 504 167 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:52:47.542149426Z stdout F 100.64.6.28 - - [22/Sep/2023:05:52:47 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:52:52.94276874Z stdout F 100.64.6.28 - - [22/Sep/2023:05:52:52 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 2386 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:53:48.029810753Z stderr F 2023/09/22 05:53:48 [error] 15#15: *34508 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "api.farm.openzim.org"
2023-09-22T05:53:48.030016578Z stdout F 100.64.6.28 - - [22/Sep/2023:05:53:48 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 504 167 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:54:49.86058916Z stderr F 2023/09/22 05:54:49 [error] 15#15: *34372 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "api.farm.openzim.org"
2023-09-22T05:54:49.860660635Z stdout F 100.64.6.28 - - [22/Sep/2023:05:54:49 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 504 167 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:55:14.092930054Z stderr F 2023/09/22 05:55:14 [warn] 15#15: *34671 a client request body is buffered to a temporary file /var/cache/nginx/client_temp/0000002436, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", host: "api.farm.openzim.org"
2023-09-22T05:55:14.182622918Z stdout F 100.64.6.28 - - [22/Sep/2023:05:55:14 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 500 265 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:55:55.775066128Z stdout F 100.64.6.28 - - [22/Sep/2023:05:55:55 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 2386 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:56:14.781624589Z stderr F 2023/09/22 05:56:14 [error] 15#15: *34334 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 100.64.6.28, server: , request: "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1", upstream: "uwsgi://unix:///tmp/uwsgi.sock", host: "api.farm.openzim.org"
2023-09-22T05:56:14.781806409Z stdout F 100.64.6.28 - - [22/Sep/2023:05:56:14 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 504 167 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:58:15.416342864Z stdout F 100.64.6.28 - - [22/Sep/2023:05:58:15 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 500 265 "-" "python-requests/2.31.0" "65.28.105.18"
2023-09-22T05:58:58.623849675Z stdout F 100.64.6.28 - - [22/Sep/2023:05:58:58 +0000] "GET /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 200 2386 "-" "python-requests/2.30.0" "65.28.105.18"
2023-09-22T05:59:16.09143754Z stdout F 100.64.6.28 - - [22/Sep/2023:05:59:16 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
```

For instance, just after the first 499, we have a container ping. It fails after 1 minute but without any message in between, it just look like the thing is not responding anymore.
```
2023-09-22T05:33:20.542235066Z stderr F INFO:common.utils:Task Container ping: 1cfdca43-873a-4b36-ab42-7baf655d72c9
2023-09-22T05:34:20.539623728Z stdout F 100.64.6.28 - - [22/Sep/2023:05:34:20 +0000] "PATCH /v1/tasks/1cfdca43-873a-4b36-ab42-7baf655d72c9 HTTP/1.1" 499 0 "-" "python-requests/2.31.0" "65.28.105.18"
```

Of course this means that we also lost all events about other uploaded files, check status, and final success state.

I faked a "container ping" at 08:24 and it also timed-out after 1 minute (with a 504, but I suspect that there is little difference between 499 and 504 errors, it is just random, depending on whether nginx or python give up first, looks like they have the same timeout setting).

From this point, I suspected that we had a DB issue where the task row is locked.

I performed a `SELECT * FROM pg_locks;` and it is clear:
- there are way too many pending pg_locks, many linked to the `tasks` table
- my query induced a lock which has not been granted (but is still there)
- what is weird is that most locks on the `tasks` table are `ExclusiveLock`

```
EXCLUSIVE (ExclusiveLock)

    Conflicts with the ROW SHARE, ROW EXCLUSIVE, SHARE UPDATE EXCLUSIVE, SHARE, SHARE ROW EXCLUSIVE, EXCLUSIVE, and ACCESS EXCLUSIVE lock modes. This mode allows only concurrent ACCESS SHARE locks, i.e., only reads from the table can proceed in parallel with a transaction holding this lock mode.

    Acquired by REFRESH MATERIALIZED VIEW CONCURRENTLY.
```

We can also analyze which queries are blocked:
```
select pid,
       usename,
       pg_blocking_pids(pid) as blocked_by,
       state as query_state,
       client_addr as client_addr,
       query as blocked_query
from pg_stat_activity
where cardinality(pg_blocking_pids(pid)) > 0;
```

This points to two root PIDs (PG PID) 168265 and 170012 which are blocking all other queries. All queries are `UPDATE task SET ...`. Not clear why those queries are induced `ExclusiveLock` ... Both are from the API (IP 100.64.6.105).

At this stage, another incident occured : the Zimfarm API got completely stuck by those locks (HTTP queue full, no more capability to respond to HTTP requests in time, exactly https://github.com/kiwix/k8s/issues/111)
- restart the API => this released all pending locks in the DB and restored the service
- issue a fake "Container ping" on our task => 204 immediately returned, no lock in the DB
- cancel the task and requested it again => https://farm.openzim.org/pipeline/ad95c9b6-9365-4972-93b5-c80fd14c2192
- open an issue regarding the improper CMS error handling => DOING
- open an issue regarding pending lock in the DB, probably linked to the exception that occured during exception handling and after a session.flush() => DOING

**Comment by benoit74** [comment]
CMS error handling => https://github.com/openzim/zimfarm/issues/825
DB locks issue => https://github.com/openzim/zimfarm/issues/826

**Comment by benoit74** [comment]
Output of `ss -xl` during the event:
```
Netid                State                 Recv-Q                Send-Q                                            Local Address:Port                                      Peer Address:Port
u_str                LISTEN                0                     1024                                 /var/run/supervisor.sock.1 1826736378                                           * 0
u_str                LISTEN                101                   100                                             /tmp/uwsgi.sock 1826736448                                           * 0
```

There has been a significant number of queries lost in the event, we will have to monitor tasks for any one where we might have lost the Succeeded or Failed events. I will look into this list later.

**Comment by benoit74** [comment]
La liste des tâches à nettoyer

| Task | Comment |
|--|--|
| [073f529b-2e99-4005-95d6-59f456d390c9](https://farm.openzim.org/pipeline/073f529b-2e99-4005-95d6-59f456d390c9)| File uploaded but not checked, final status lost, cancelled and re-requested |
| [0d0bcd67-e516-40db-b2c3-38b8703678c0](https://farm.openzim.org/pipeline/0d0bcd67-e516-40db-b2c3-38b8703678c0)| OK, status failed is recorded |
| [1703794e-55b5-4680-82af-ead623feae2d](https://farm.openzim.org/pipeline/1703794e-55b5-4680-82af-ead623feae2d)| OK, still running |
| [1cfdca43-873a-4b36-ab42-7baf655d72c9](https://farm.openzim.org/pipeline/1cfdca43-873a-4b36-ab42-7baf655d72c9) | "THE" task, cancelled and re-requested |
| [43828414-e124-4bee-993c-9e7e5a2f8475](https://farm.openzim.org/pipeline/43828414-e124-4bee-993c-9e7e5a2f8475)| OK, status succeeded + files uploaded and checked |
| [49605fc7-c2e2-4960-89cd-3a5ed26a2762](https://farm.openzim.org/pipeline/49605fc7-c2e2-4960-89cd-3a5ed26a2762)| OK, still running (after an erroneous manual PATCH to success + tedious manual rollback in DB) |
| [5e2242be-e95a-457f-b93a-dafc26d0f57b](https://farm.openzim.org/pipeline/5e2242be-e95a-457f-b93a-dafc26d0f57b)| OK, status succeeded recorded  |
| [95c16592-3a1c-40f1-ae07-c253f950e0d1](https://farm.openzim.org/pipeline/95c16592-3a1c-40f1-ae07-c253f950e0d1)| OK, status failed recorded |
| [95d84941-c584-488d-9483-57b769b4322d](https://farm.openzim.org/pipeline/95d84941-c584-488d-9483-57b769b4322d)| OK, still running |
| [9cf664a1-bf9d-4ab3-a498-f716c6f8d473](https://farm.openzim.org/pipeline/9cf664a1-bf9d-4ab3-a498-f716c6f8d473)| OK, status failed recorded |
| [9db6bdb3-5543-410d-a991-439d71138512](https://farm.openzim.org/pipeline/9db6bdb3-5543-410d-a991-439d71138512)| OK, still running |
| [ad95c9b6-9365-4972-93b5-c80fd14c2192](https://farm.openzim.org/pipeline/ad95c9b6-9365-4972-93b5-c80fd14c2192)| OK, still running |
| [b03d162d-94c0-40c7-8785-a880220b4a2e](https://farm.openzim.org/pipeline/b03d162d-94c0-40c7-8785-a880220b4a2e)| OK, still running |
| [b0aed843-faaf-47de-ba75-febb9af17418](https://farm.openzim.org/pipeline/b0aed843-faaf-47de-ba75-febb9af17418)| OK, still running |
| [c7d1c21a-bb44-40cc-ba3e-4ba95cbe6f8e](https://farm.openzim.org/pipeline/c7d1c21a-bb44-40cc-ba3e-4ba95cbe6f8e)| OK, still running |
| [d12b352d-0498-47f1-904f-977562caaf0a](https://farm.openzim.org/pipeline/d12b352d-0498-47f1-904f-977562caaf0a)| OK, still running |
| [d2fed3b1-1fd5-4b65-8d60-4896bdfe2693](https://farm.openzim.org/pipeline/d2fed3b1-1fd5-4b65-8d60-4896bdfe2693)| OK, still running |
| [e04bb64b-4a96-41b5-bdb1-1050faa0c2af](https://farm.openzim.org/pipeline/e04bb64b-4a96-41b5-bdb1-1050faa0c2af)| OK, still running |
| [e0fdb248-5663-4a7d-abc9-d0f9c55daed2](https://farm.openzim.org/pipeline/e0fdb248-5663-4a7d-abc9-d0f9c55daed2)| OK, status succeeded recorded |

**Comment by benoit74** [comment]
Closing this issue as no more work is needed on this task and siblings + other issues have been opened and are active.
